### PR TITLE
add support debian buster and remove some ansible warnings

### PR DESCRIPTION
**Pull request by sv99**
1. debian Buster
2. user create with bash shell
3. remove ansible warning

### CODE REVIEW

**Comment by nickjj** [comment]
Hi,

What tools / Ansible version did you use to get those Ansible warnings?

**Comment by sv99** [comment]
2 task not have | d()!! 

``` bash
ansible --version
ansible 2.8.2
  config file = /Users/svolkov/.ansible/ansible.cfg
  configured module search path = ['/Users/svolkov/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/svolkov/.pyenv/versions/3.6.3/lib/python3.6/site-packages/ansible
  executable location = /Users/svolkov/.pyenv/versions/3.6.3/bin/ansible
  python version = 3.6.3 (default, Nov  8 2017, 19:00:49) [GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.37)]
```

File: tasks/main.yml [hunk 1]
```diff
@@ -18,10 +18,11 @@
 - name: Install Docker's dependencies
   apt:
     name: "{{ docker__package_dependencies + docker__pip_dependencies }}"
+    force_apt_get: true
```
Comment (nickjj) [inline tasks/main.yml]: Can you remove this?

This seems to be an Ansible bug that will be addressed in future 2.8.x patches: https://github.com/ansible/ansible/issues/56832

File: tasks/main.yml [hunk 2]
```diff
@@ -18,10 +18,11 @@
...
 
 - name: Add Docker's public PGP key to the APT keyring
   apt_key:
-    id: "{{ docker__apt_key_id }}"
+    #    id: "{{ docker__apt_key_id }}"
```
Comment (nickjj) [inline tasks/main.yml]: What was your reasoning behind removing this?

File: tasks/main.yml [hunk 3]
```diff
@@ -81,6 +83,7 @@
     name: "{{ item }}"
     groups: "docker"
     append: true
+    shell: "/bin/bash"
```
Comment (nickjj) [inline tasks/main.yml]: Can you remove this? This isn't necessary to set.

File: tasks/main.yml [hunk 4]
```diff
@@ -101,7 +104,7 @@
     owner: "root"
     group: "root"
     mode: "0644"
-  when: docker__default_daemon_json or docker__daemon_json
+  when: (docker__default_daemon_json | d()) or (docker__daemon_json | d())
```
Comment (nickjj) [inline tasks/main.yml]: Looks good.

File: tasks/main.yml [hunk 5]
```diff
@@ -112,7 +115,8 @@
     group: "root"
     mode: "0644"
   register: docker__register_daemon_flags
-  when: docker__daemon_flags
+  when: docker__daemon_flags | d()
+
```
Comment (nickjj) [inline tasks/main.yml]: Can you get rid of the extra line break here?

**Comment by nickjj** [comment]
I see. I just updated to Ansible 2.8.x and noticed them too.

Besides the recommended changes listed above can you also break this up into 2 separate commits?

You can name them like this:

- Officially support Debian Buster
- Fix Ansible 2.8.x warnings

File: tasks/main.yml (hunk 2 above)
Comment (sv99) [reply tasks/main.yml]: id in this case is excessive, id need for apt key server

**Pull request closed without merging**